lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: cyberark_authentication
+short_description: "Module for CyberArk Vault Authentication using Privileged Account Security Web Services SDK"
+author: "Edward Nunez (@enunez-cyberark)"
+version_added: "2.3"
+description:
+    - "Authenticates to CyberArk Vault using Privileged Account Security Web Services SDK and
+       creates a session fact that can be used by other modules. It returns an Ansible fact
+       called I(cyberarkSession). Every module can use this fact as C(cyberarkSession) parameter."
```

Comment:
I took a look at cloudstack modules and module_utils, and I see the need for it based on using a python library and having that as a utility module for it. 

In this case, the functionality of the cyberark modules is based on REST apis available accross the board. 
I based it the same way it was done for module ovirt_auth under modules/cloud/ovirt